src/main/java/mate/dao/DriverDaoImpl.java

```diff
@@ -101,11 +105,31 @@ public boolean delete(Long id) {
         }
     }
 
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT * FROM drivers WHERE login = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = getDriver(resultSet);
+                return Optional.of(driver);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find driver by login " + login, e);
+        }
+        return Optional.empty();
```

Comment:
there is no need for such complication
```suggestion
            Driver driver = null;
            if (resultSet.next()) {
                driver = getDriver(resultSet);
            }
            return Optional.ofNullable(driver);
        } catch (SQLException e) {
            throw new DataProcessingException("Can't find driver by login " + login, e);
        }
```
//or
```suggestion
            if (resultSet.next()) {
                Driver driver = getDriver(resultSet);
                return Optional.of(driver);
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't find driver by login " + login, e);
        }
        return Optional.empty();
```